internal/dialect/dialectquery/clickhouse.go

```diff
@@ -42,3 +42,44 @@ func (c *Clickhouse) GetLatestVersion(tableName string) string {
 	q := `SELECT max(version_id) FROM %s`
 	return fmt.Sprintf(q, tableName)
 }
+
+type ClickhouseReplicated struct{}
+
+var _ Querier = (*ClickhouseReplicated)(nil)
+
+func (c *ClickhouseReplicated) CreateTable(tableName string) string {
+	q := `CREATE TABLE IF NOT EXISTS %s ON CLUSTER '{cluster}' (
```

Comment:
Hi @funvit! Thanks for the feedback 👍

I understand the concern about the `{cluster}` macro not being a standard. While it might not be universally recognized, it is widely used when deploying clusters via the [Altinity ClickHouse Operator](https://github.com/Altinity/clickhouse-operator/blob/d5f265fb4773ec4622b2af13fa52858c9f1e8c15/docs/replication_setup.md#macros), and it’s also mentioned in the official [ClickHouse documentation](https://clickhouse.com/docs/en/sql-reference/table-functions/cluster).

I believe using the `{cluster}` macro is a good practice because it simplifies cluster management and cluster queries.

Regarding making it configurable, the reason I didn’t pursue that route is due to the current limitations in the implementation of dialects, which don’t support additional settings. To avoid significant changes, something I’d prefer to minimize when adding a new dialect, the most viable approach is to pass the cluster name via an environment variable, like `CLICKHOUSE_REPLICATED_CLUSTER_NAME`, defaulting to `{cluster}`.

What do you think - should we keep using the macro as is, or switch to an environment variable? Or maybe there's a better solution I haven't thought of?

Reply:
Perhaps this macro is standard or recommended, but in tests it would be easier to replace it with some custom value.

For example, now I had to implement Store for an older version of goose, and there I just made a Cluster field for ClickhouseReplicated.

This simplified the tests (using testcontainers), because it was no longer necessary to support the configuration of macros.

Reply:
Thanks for explanation, it makes sense!

I'm added extra env var `GOOSE_CLICKHOUSE_CLUSTER_NAME` which allows redefine cluster name and added `clickhouse-replicated` dialect tests. PTAL

Reply:
friendly ping :)

Reply:
LGTM